stores/shoppingCart.ts

```diff
@@ -46,37 +38,33 @@ export const useShoppingCartStore = defineStore('shoppingCart', {
       this.itemToBuy = undefined
     },
     updateItem(payload: ShoppingCartItem) {
-      const existinItemIndex = items().findIndex(
-        (item) => item.id === payload.id
-      )
-      if (existinItemIndex === -1) {
-        return
-      } else {
-        const items = [...localStorage.value]
-        items[existinItemIndex] = {
-          ...this.items[existinItemIndex],
+      const itemIndex = existInItemIndex(payload.id, this.items)
+      if (itemIndex !== -1) {
+        this.items[itemIndex] = {
+          ...this.items[itemIndex],
           ...payload,
         }
-        localStorage.value = items
+        localStorage.value = this.items
       }
     },
     removeItem(id: ID) {
-      const existinItemIndex = items().findIndex((item) => item.id === id)
-      if (existinItemIndex === -1) {
-        return
-      } else {
-        const items = [...localStorage.value]
-        items.splice(existinItemIndex, 1)
-        localStorage.value = items
+      const itemIndex = existInItemIndex(id, this.items)
+      if (itemIndex !== -1) {
+        this.items.splice(itemIndex, 1)
+        localStorage.value = this.items
       }
     },
     clear() {
       localStorage.value = []
       this.itemToBuy = undefined
       this.items = []
     },
-    setItems(payload: ShoppingCartItem[]) {
-      localStorage.value = payload
+    toggleItemInCart(nft: ShoppingCartItem) {
+      if (this.isItemInCart(nft.id)) {
+        this.removeItem(nft.id)
+      } else {
+        this.setItem(nft)
+      }
```

Comment:
I don't see any other of our stores having 'toggle'
Usually, stores do get and set, and toggle is business logic better implemented by the component.
especially since it was used only in one place prior to this refactor

Reply:
You're correct we can put it back. I was just trying to encapsulate all that relates to mutation of this store i add as an action. We can also provide a better name for this instead of toggle